modules/scratchFetch.ts

```diff
@@ -0,0 +1,242 @@
+import constants from "../common/constants.js";
+
+defineEvent("messageCreate", async (message) => {
+	const notSet = getSettings(message.author, false)?.scratchEmbeds === undefined;
+	if (!getSettings(message.author).scratchEmbeds) {
+		return;
+	}
+	const scratchUrlRegex =
+		/(?<!<)https?:\/\/scratch\.mit\.edu\/(projects|users|studios|discuss)\/[/-\w]+\/?(?!>)/; //gpt wrote the regex and like half of this code
+	const match = message.content.match(scratchUrlRegex);
+
+	if (!match) {
+		return;
+	}
+
+	const urlParts = match[0].split("/");
+	const type = urlParts[3];
+
+	async function fetchApiData(apiUrl: string): Promise<any> {
+		try {
+			const response = await fetch(apiUrl);
+
+			const data = await response.json();
+			return data;
+		} catch (error) {}
+	}
+	function long(text: string, maxLength: number, appendText: string): string {
+		if (text.length > maxLength) {
+			return text.substr(0, maxLength) + appendText;
+		}
+		return text;
+	}
+	switch (type) {
+		case "projects":
+			const projectdata = await fetchApiData(
+				`https://${constants.urls.scratchApi}/projects/${urlParts[4]}/`,
+			).catch(() => {});
+
+			message.reply({
+				embeds: [
+					{
+						title: projectdata.title,
+						description: `**desc**: ${projectdata.description}\n**inst**: ${projectdata.instructions}`,
+						color: 0x885cd4,
+
+						fields: [
+							{
+								name: `views`,
+								value: projectdata.stats.views,
+								inline: true,
+							},
+							{
+								name: `loves`,
+								value: projectdata.stats.loves,
+								inline: true,
+							},
+							{
+								name: `favorites`,
+								value: projectdata.stats.favorites,
+								inline: true,
+							},
+							{
+								name: `remixes`,
+								value: projectdata.stats.remixes,
+								inline: true,
+							},
+						],
+						thumbnail: {
+							url: projectdata.images["282x218"],
+						},
+						author: {
+							name: projectdata.author.username,
+							url: `https://scratch.mit.edu/users/${projectdata.author.username}`,
+							icon_url: projectdata.author.profile.images["90x90"],
+						},
+						footer: {
+							text: notSet ? "Disable this using /settings" : "",
+						},
+						url: `https://scratch.mit.edu/projects/${urlParts[4]}`,
+					},
+				],
+				allowedMentions: { repliedUser: false },
+			});
+
+			break;
+		case "users":
+			const userdata = await fetchApiData(
+				`https://${constants.urls.scratchApi}/users/${urlParts[4]}/`,
+			).catch(() => {});
+
+			message.reply({
+				embeds: [
+					{
+						title: userdata.username,
+						description: "",
+						color: 0x885cd4,
+
+						fields: [
+							{
+								name: `About`,
+								value: userdata.profile.bio,
+							},
+							{
+								name: `WIWO`,
+								value: userdata.profile.status,
+							},
+						],
+						thumbnail: {
+							url: userdata.profile.images["90x90"],
+						},
+						author: {
+							name: userdata.scratchteam ? "Scratch Team" : "",
```

Comment:
put the country here
if they're a new scratcher, also put `${constants.footerSeperator}New Scratcher` after it